module5/TaskListApp/TaskListApp/Views/AddTaskView.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  AddTaskView.swift
+//  TaskListApp
+//
+//  Created by Peter Rice on 12/22/23.
+//
+
+import SwiftUI
+
+struct AddTaskView: View {
+  var tasksStore: TasksStore
+  @State var taskTitle: String = ""
+  @State var taskNotes: String = ""
+  @Binding var showingAddTaskView: Bool
+  
+  var body: some View {
+    NavigationView {
+      VStack {
+        Form {
+          Section(header: Text("Task Title"))
+          {
+            TextField("Title", text: $taskTitle)
+          }
+          Section(header: Text("Notes")) {
+            TextField("Notes", text: $taskNotes, axis: .vertical)
+              .lineLimit(5...)
+          }
+        }
+      }
+      .navigationBarTitle(Text("Add New Task"), displayMode: .inline)
+      .navigationBarItems(
+        leading: Button(action: {
+          showingAddTaskView = false
+        }, label: {
+          Text("Cancel")
+        }),
+        trailing: Button(action: {
+          tasksStore.addTask(title: taskTitle, notes: taskNotes)
+          showingAddTaskView = false
```

Comment:
You will aslo see a special environment variable used in cases like this `@Environment(\.dismiss) var dismiss` and then you just `dismiss()` to dismiss the view. Either way is fine. Using the environment variable keeps you from having to pass a binding variable into your view.

Reply:
Nice, this is cleaner.